SymlinkUtility.cs

```diff
@@ -7,119 +7,221 @@
+            Symlink(true);
+        }
+#endif
+
+        static void Symlink(bool absolute)
+        {
+            string sourceFolderPath = EditorUtility.OpenFolderPanel("Select Folder Source", "", "");
+
+            // Cancelled dialog
+            if (string.IsNullOrEmpty(sourceFolderPath))
+                return;
+
+            if (sourceFolderPath.Contains(Application.dataPath))
+            {
+                UnityEngine.Debug.LogWarning("Cannot create a symlink to folder in your project!");
+                return;
+            }
+
+            string sourceFolderName = sourceFolderPath.Split(new char[] { '/', '\\' }).LastOrDefault();
+
+            if (string.IsNullOrEmpty(sourceFolderName))
+            {
+                UnityEngine.Debug.LogWarning("Couldn't deduce the folder name?");
+                return;
+            }
+
+            Object uobject = Selection.activeObject;
+
+            string targetPath = uobject != null ? AssetDatabase.GetAssetPath(uobject) : null;
+
+            if (string.IsNullOrEmpty(targetPath))
+                targetPath = "Assets";
+
+            FileAttributes attribs = File.GetAttributes(targetPath);
+
+            if ((attribs & FileAttributes.Directory) != FileAttributes.Directory)
+                targetPath = Path.GetDirectoryName(targetPath);
+
+            // Get path to project.
+            string pathToProject = Application.dataPath.Split(new string[1] { "/Assets" }, System.StringSplitOptions.None)[0];
+
+            targetPath = string.Format("{0}/{1}/{2}", pathToProject, targetPath, sourceFolderName);
+
+            if (Directory.Exists(targetPath))
+            {
+                UnityEngine.Debug.LogWarning(string.Format("A folder already exists at this location, aborting link.\n{0} -> {1}", sourceFolderPath, targetPath));
+                return;
+            }
+
+#if UNITY_EDITOR_WIN
+			using (Process cmd = Process.Start("CMD.exe", string.Format("/C mklink /J \"{0}\" \"{1}\"", targetPath, sourceFolderPath)))
+            {
+                cmd.WaitForExit();
+            }
+#elif UNITY_EDITOR_OSX
+            // For some reason, OSX doesn't want to create a symlink with quotes around the paths, so escape the spaces instead.
+            sourceFolderPath = sourceFolderPath.Replace(" ", "\\ ");
+            targetPath = targetPath.Replace(" ", "\\ ");
+            // Use absolute path or relative path?
+            string sourcePath = absolute ? sourceFolderPath : GetRelativePath(sourceFolderPath, targetPath);
+            string command = string.Format("ln -s {0} {1}", sourcePath, targetPath);
+            ExecuteBashCommand(command);
+#elif UNITY_EDITOR_LINUX
+            // Is Linux the same as OSX?
+#endif
+
+            //UnityEngine.Debug.Log(string.Format("Created symlink: {0} <=> {1}", targetPath, sourceFolderPath));
+
+            AssetDatabase.Refresh(ImportAssetOptions.ForceUpdate);
+        }
+
+        static string GetRelativePath(string sourcePath, string outputPath)
```

Comment:
Could this use https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getrelativepath?view=netcore-3.1 ?

Reply:
I also found that after I had already written this, haha. Unfortunately, that is not available in .Net Standard 2.0 (which is what Unity is currently on). Hopefully we'll get a lot more .Net functionality with the .Net 5.0 project!